### ISSUE TITLE
Unbounded class size bug

### ISSUE THREAD

**Issue by sei-eschwartz** [cmu-sei/pharos#102]
Both V9.exe and BattleForge.exe got stuck in a seemingly endless cycle similar to this:

```
Concluding factClassSizeGTE(0x17e73c0, 0xb8a780).
Concluding factClassSizeGTE(0x13e3b50, 0xb8a76c).
Concluding factClassSizeGTE(0x443110, 0xb8a754).
Concluding factClassSizeGTE(0x442fd0, 0xb8a754).
Concluding factClassSizeGTE(0x17e78f0, 0xb8a9d0).
reasonForwardAsManyTimesAsPossible
Starting reasonForward.
Concluding factClassSizeGTE(0x16aa570, 0xb8a780).
Concluding factClassSizeGTE(0x16aa510, 0xb8a780).
Concluding factClassSizeGTE(0x129c490, 0xb8a79c).
Concluding factClassSizeGTE(0x145cc00, 0xb8a76c).
Concluding factClassSizeGTE(0x1459080, 0xb8a76c).
Concluding factClassSizeGTE(0x145f240, 0xb8a76c).
```

Our first thought is that our cycle detection is not good enough, specifically probably that relating to embedded objects.

**Comment by sei-eschwartz** [comment]
@janbbeck and @xnxexox you should probably be interested in this, because it affects your .exes.

**Comment by janbbeck** [comment]
Thanks for the heads up! I did notice this, but am not in a position to tell if its expected or not. This has been running on my computer for several days now, spitting out those messages :P

**Comment by sei-eschwartz** [comment]
I am pretty sure we fixed this, but I am not sure how long it took to trigger the problem originally, so I can't say I've confirmed it (yet).

**Comment by sei-eschwartz** [comment]
@janbbeck V9.exe now completed for me
[V9.exe.results.gz](https://github.com/cmu-sei/pharos/files/4764808/V9.exe.results.gz)

**Comment by janbbeck** [comment]
Thanks!

**Comment by janbbeck** [comment]
Does the latest docker have this fix yet?

**Comment by sei-eschwartz** [comment]
> Does the latest docker have this fix yet?

No, I manually started this now, and in the future it should start building automatically within an hour.

**Comment by janbbeck** [comment]
This does not work for me. Reproduced on Windows and Linux:
- install latest plugin (ghidra_9.1.2_PUBLIC_20200613_OOAnalyzerPlugin.zip) from : https://github.com/sei-eschwartz/pharos/releases
- load v9.exe
- perform analysis
- got to CERT->ooanalyzer and load the file


![Screenshot from 2020-06-14 05-42-42](https://user-images.githubusercontent.com/11450961/84591196-1bee6f80-ae02-11ea-94d8-a159d007fb07.png)

**Comment by sei-eschwartz** [comment]
I just tried this and it worked for me.  The file I uploaded was gzipped; did you ungzip it?

There should be a log file on your Linux machine at `~/.ghidra/.ghidra_9.1.2_PUBLIC/application.log`.  Can you upload the most recent portion?  It will hopefully let us see what is going wrong.

**Comment by janbbeck** [comment]
I did unzip, the file starts with:
```
{
	"architecture" : 
	{
		"bitSize" : 32,
		"endian" : "little",
		"name" : "x86"
	},
```
right?

I am attaching the application log
[application.log](https://github.com/cmu-sei/pharos/files/4776342/application.log)


Thanks!

**Comment by sei-eschwartz** [comment]
I am talking about the `V9.exe.json.gz` file.  Did you ungzip that?  (That is not what it starts with)

Here is the md5sum of the uncompressed file:
`2b53033ba8ffec9490a37492beffbb8d  V9.exe.json`

**Comment by janbbeck** [comment]
ah!
looks like ghidra makes its own v9.exe.json file!

**Comment by janbbeck** [comment]
and in deed that was the problem. Sorry for the noise!

**Comment by sei-eschwartz** [comment]
No problem :-)

**Comment by sei-eschwartz** [comment]
@xNxExOx I was able to finish reasoning for your file.  It took several days, and unfortunately my run did not include a fix for #101 so I can't easily produce json from it.  But you could probably hand edit the .results file to remove the duplicate keys and use ooJson.
[BattleForge.exe.results.gz](https://github.com/cmu-sei/pharos/files/4786385/BattleForge.exe.results.gz)
